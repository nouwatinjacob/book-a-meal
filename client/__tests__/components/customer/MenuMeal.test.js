import React from 'react';
import {
  shallow
} from 'enzyme';
import {
  MenuMeal,
  mapStateToProps
} from '../../../src/components/customer/MenuMeal.jsx';
import {
  retrieveMenuResponse
} from '../../__mockData__/mockMenu';

const props = {
  getMenuAction: jest.fn(),
  menuState: {
    menus: retrieveMenuResponse
  }
};


describe('MenuMeals component', () => {
  localStorage.setItem('token', customerToken);
  const setup = () => shallow(<MenuMeal {...props} />);

  it('should render MenuMeals component correctly', (done) => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();

    done();
  });

  it(`should call checkYesterdayMenu when
  checking if meal is yesterday menu`, (done) => {
    const wrapper = setup();

    const checkYesterdayMenuSpy = jest.spyOn(
      wrapper.instance(),
      'checkYesterdayMenu'
    );
    wrapper.instance().checkYesterdayMenu();
    expect(checkYesterdayMenuSpy).toHaveBeenCalled();

    done();
  });

  it(`should call 'handlePageClick'
  if pagination button is clicked`, (done) => {
    const wrapper = setup();

    const data = {
      selected: 1
    };

    const handlePageClickSpy = jest.spyOn(
      wrapper.instance(),
      'handlePageClick'
    );
    wrapper.instance().handlePageClick(data);
    expect(handlePageClickSpy).toHaveBeenCalled();

    done();
  });

  it(`should call 'onClickOrder'
  if order meal button if clicked`, (done) => {
    const wrapper = setup();

    const mealId = 1;
    const menuId = 1;
    const mealUserId = 1;
    const userId = 2;

    const onClickOrderSpy = jest.spyOn(
      wrapper.instance(),
      'onClickOrder'
    );
    wrapper.instance().onClickOrder(mealId, menuId, mealUserId, userId);
    expect(onClickOrderSpy).toHaveBeenCalled();
    done();
  });

  it(`should call 'handleDateChange'
  if date is change on the calendar`, (done) => {
    const wrapper = setup();

    const event = {
      target: {
        value: '5'
      }
    };

    const handleDateChangeSpy = jest.spyOn(
      wrapper.instance(),
      'handleDateChange'
    );
    wrapper.instance().handleDateChange(event);
    expect(handleDateChangeSpy).toHaveBeenCalled();

    done();
  });

  it(`should call componentWillUnmount if
   unmount component `, (done) => {
    const wrapper = setup();

    const componentWillUnmountSpy = jest.spyOn(
      wrapper.instance(),
      'componentWillUnmount'
    );
    wrapper.instance().componentWillUnmount();
    expect(componentWillUnmountSpy).toHaveBeenCalled();

    done();
  });
});