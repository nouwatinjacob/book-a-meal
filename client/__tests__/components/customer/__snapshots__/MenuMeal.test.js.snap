// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuMeals component should render MenuMeals component correctly 1`] = `
<div>
  <div
    className="container"
  >
    <CatererHeader />
    <div
      className="wrapper"
    >
      <div
        className="wrapper search"
      >
        <label>
          Search by date
        </label>
        <input
          onChange={[Function]}
          type="date"
        />
      </div>
      <div
        className="row"
      >
        <div
          className="c-medium-3 c-xsmall-12 c-3"
          id="pd-0"
          key="0"
        >
          <div
            className="box"
          >
            <div
              id="menu-image"
            >
              <img
                alt="Avatar"
                src="https://res.cloudinary.com/sansaristic/image/upload/v1536197477/BookMeal/1536197474332pexels-photo-247685.png.png"
              />
            </div>
            <div
              className="box-body"
            >
              <div
                className="row"
              >
                <div
                  className="c-medium-12 c-xsmall-12 text-center"
                  id="pd-0"
                >
                  <p>
                    <i>
                      Stephen's Shop
                    </i>
                  </p>
                  <p>
                    <b>
                      Fried Rice and chicken Lap
                    </b>
                  </p>
                  <p>
                    Price 
                    3000
                  </p>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="c-medium-12 c-xsmall-12 text-center"
                  id="pd-0"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="c-medium-3 c-xsmall-12 c-3"
          id="pd-0"
          key="1"
        >
          <div
            className="box"
          >
            <div
              id="menu-image"
            >
              <img
                alt="Avatar"
                src="https://res.cloudinary.com/sansaristic/image/upload/v1536198430/BookMeal/1536198429005images.jpeg.jpg"
              />
            </div>
            <div
              className="box-body"
            >
              <div
                className="row"
              >
                <div
                  className="c-medium-12 c-xsmall-12 text-center"
                  id="pd-0"
                >
                  <p>
                    <i>
                      Stephen's Shop
                    </i>
                  </p>
                  <p>
                    <b>
                      Ewa Aganhin with Fish
                    </b>
                  </p>
                  <p>
                    Price 
                    2500
                  </p>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="c-medium-12 c-xsmall-12 text-center"
                  id="pd-0"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="c-medium-3 c-xsmall-12 c-3"
          id="pd-0"
          key="2"
        >
          <div
            className="box"
          >
            <div
              id="menu-image"
            >
              <img
                alt="Avatar"
                src="https://res.cloudinary.com/sansaristic/image/upload/v1536198463/BookMeal/1536198460052pexels-photo-247685.png.png"
              />
            </div>
            <div
              className="box-body"
            >
              <div
                className="row"
              >
                <div
                  className="c-medium-12 c-xsmall-12 text-center"
                  id="pd-0"
                >
                  <p>
                    <i>
                      Stephen's Shop
                    </i>
                  </p>
                  <p>
                    <b>
                      Garri
                    </b>
                  </p>
                  <p>
                    Price 
                    2000
                  </p>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="c-medium-12 c-xsmall-12 text-center"
                  id="pd-0"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <br />
    </div>
  </div>
  <br />
</div>
`;
